packages/unified-selection/src/unified-selection/iModel/EnableUnifiedSelectionSyncWithIModel.ts

```diff
@@ -0,0 +1,46 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+
+/** @packageDocumentation
+ * @module UnifiedSelection
+ */
+
+import { CachingHiliteSetProvider } from "../CachingHiliteSetProvider";
+import { IMetadataProvider } from "../queries/ECMetadata";
+import { IECSqlQueryExecutor } from "../queries/ECSqlCore";
+import { ElementSelectionScopeProps, SelectionScope } from "../SelectionScope";
+import { SelectionStorage } from "../SelectionStorage";
+import { IModelConnection } from "./IModel";
+import { IModelSelectionHandler } from "./IModelSelectionHandler";
+
+/**
+ * Props for `enableUnifiedSelectionSyncWithIModel`.
+ * @internal Not exported through barrel, but used in public API as an argument. May be supplemented with optional attributes any time.
+ */
+export interface EnableUnifiedSelectionSyncWithIModelProps {
+  /** iModel to synchronize selection for. */
+  iModel: IModelConnection;
+  /** SelectionStorage to synchronize selection with. */
+  selectionStorage: SelectionStorage;
+  /** Caching hilite set provider. */
+  cachingHiliteSetProvider: CachingHiliteSetProvider;
```

Comment:
I think often `enableUnifiedSelectionSyncWithIModel` is going to be the only user of `CachingHiliteSetProvider`. Do you think we could make providing it optional? I understand creating a `CachingHiliteSetProvider` requires a function to get `queryExecutor` and `IMetadataProvider` based on imodel key, which we can't provide. But we have the required `queryExecutor` and `IMetadataProvider`, so maybe we can come up with something?

Reply:
Made it optional by using a function that returns the passed `queryExecutor` and `IMetadataProvider` when creating `CachingHiliteSetProvider`.